### PR TITLE
Docs: kpt migration documentation

### PR DESCRIPTION
**Pull request by phanimarupaka**
Here is the migration documentation for kpt 1.0.

### CODE REVIEW

File: site/migration/01-migration.md
Comment (mikebz) [inline site/migration/01-migration.md:55]: I'd just call this Alpha1 Commands

File: site/migration/01-migration.md
Comment (mikebz) [inline site/migration/01-migration.md:61]: slight language change "Deprecated in favor of reading Kptfile directly" seems less of a command.

File: site/migration/01-migration.md
Comment (mikebz) [inline site/migration/01-migration.md:104]: Wording suggestion:
'sync' functionality has been moved to the 'update` command.

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:193]: do you add the starlark function instead?

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:1]: It may be better to have this document live in `docs/` instead of `site/`. We can link to this doc from installation page and the fix command instead of having a new top-level section in the navbar.

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:72]: preview and diff have been removed as well.

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:102]: To better organize the content to make it easier to navigate. I suggest having the command groups as headers (e.g. `pkg`, `cfg`, `fn`, `live`). ToC should mirror the the command hierarchy accordingly.

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:237]: I would reword this: The package format schema will be granduated to v1.
The semver of the CLI binary can be orthogonal to this. In other words, we can guarantee backwards compatibility of package format earlier than 1.0 release of the CLI.

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:165]: In each section refer to relevant sections of kpt.dev (reference or book) for more information.

**Review by frankfarzan — COMMENTED** [review]
Please wait for @droot and @mortent to review as well.

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:105]: You may want to address the fact the Kptfile doesn't have `dependencies` section. What's the equivalent flow?

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:29]: Somewhere at the top, link to the kpt book. Specifically, the user needs to read Chapter 1 and 2, i.e. the Quickstart and the concepts, specifically the workflows (https://kpt.dev/book/02-concepts/02-workflows) to be able to make sense of things. Trying to follow the migration doc (dealing with complexity of two versions) without understanding the basic middle of new version will lead to confusion.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md:105]: We need to have a separate doc for that. Once that is ready we can link it here https://github.com/GoogleContainerTools/kpt/issues/1941

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Used version numbers instead as we are using it all across the docs. Users might not have observed v1alpha1 in Kptfile at all.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Update.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md:113]: Good point. Update to `sync` merged with `update`

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md:204]: No. Added it to manual portion.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Updated.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md:1]: Will do that in next PR.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Updated.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md:248]: I feel the wording you suggested is complex. I am having trouble understanding it.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Updated.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md:31]: Updated Before you begin section.

**Comment by phanimarupaka** [comment]
@frankfarzan apologies for force push. Did by mistake.

**Review by mikebz — APPROVED** [review]
let's merge for fishfood.

File: site/migration/01-migration.md
Comment (frankfarzan) [reply site/migration/01-migration.md:248]: There are two separate independent releases: The package format (ie.g. Kptfile and other CRD) can be graduated to apiVersion v1 *before* we release CLI version 1.0.

File: site/migration/01-migration.md
Comment (frankfarzan) [reply site/migration/01-migration.md:1]: Let's do this in this PR, otherwise the navbar will be affected in the meantime.

File: site/migration/01-migration.md
Comment (frankfarzan) [inline site/migration/01-migration.md:3]: In the GitHub preview, these don't show up as links. Not sure if that's a real issue or not.

**Review by frankfarzan — COMMENTED** [review]
Looking good. I like the organization now.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md:248]: So we have different timeline for both ? Can you make please a separate PR using edit icon after this PR is merged?

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md]: It is not possible to point to files outside the site folder from the files within site folder. So moved the folder to installation folder and added a link in installation guide. Remove it from sidebar to not pollute the view.

File: site/migration/01-migration.md
Comment (phanimarupaka) [reply site/migration/01-migration.md]: They do for me. Try this out https://github.com/GoogleContainerTools/kpt/blob/51e2495577f27c272210dcd8a17f3090797f2f91/site/installation/migration/01-migration.md#Before-you-begin

File: site/installation/README.md
Comment (frankfarzan) [inline site/installation/README.md:3]: To future proof this, reword:

If you're currently using version v0.39, follow the ... to migrate.

File: site/installation/README.md
Comment (frankfarzan) [inline site/installation/README.md:100]: you can move to /docs and then link using

https://github.com/GoogleContainerTools/kpt/blob/next/docs/migration.md

File: site/installation/migration/01-migration.md
Comment (frankfarzan) [inline site/installation/migration/01-migration.md:1]: if you're keeping in /site not sure why it can't be in `installation/migration.md`

**Review by frankfarzan — APPROVED** [review]
/lgtm with some comments.

**Pull request merged**
